snapcraft/_store.py

```diff
@@ -417,27 +431,90 @@ def push(snap_filename, release_channels=None):
     with _requires_login():
         store.push_precheck(snap_name)
 
-    with _requires_login():
-        tracker = store.upload(snap_name, snap_filename)
+    snap_cache = cache.SnapCache(project_name=snap_name)
+    source_snap = snap_cache.get_latest(snap_name)
+
+    sha3_384_available = hasattr(hashlib, 'sha3_384')
+
+    if os.environ.get('DELTA_UPLOADS_EXPERIMENTAL') and \
+       sha3_384_available and source_snap:
```

Comment:
Personally I prefer parentheses to escaped newlines, and that matches the rest of the codebase.